packages/babel-parser/src/parser/statement.js

```diff
@@ -237,7 +237,7 @@ export default class StatementParser extends ExpressionParser {
   // ImportDeclaration and ExportDeclaration are also handled here so we can throw recoverable errors
   // when they are not at the top level
   parseStatement(context: ?string, topLevel?: boolean): N.Statement {
-    if (this.match(tt.at)) {
+    if (this.match(tt.at) || this.match(tt.atInit)) {
```

Comment:
We could have an helper function `tokenIsDecorator(this.state.type)` which checks for both, similarly to all the helpers we already have in https://github.com/babel/babel/blob/3b6f061e18b54cb648814f8c5bbe49472a9fe9b4/packages/babel-parser/src/tokenizer/types.js#L331.